### ISSUE TITLE
Undefined behavior leads to different program outputs when compiled with GCC 4-6 vs GCC 7

### ISSUE THREAD

**Issue by unode** [hyattpd/Prodigal#34]
When compiling Prodigal with GCC versions 4 to 6 the output produced is missing some information in the header of each sequence when compared with a Prodigal version compiled with GCC 7.

The incorrect output is (GCC versions 4.9.3 and 6.4.0 (alpine linux)):
```
>0_1 # 1 # 1344 # 1 # ;gc_cont=0.612
GTGGATAGCGGAGAGGTGAATCACTTGGAAGACCTGAAGGTCCTGTGGAACGAGATACTGGACAAGGGCA
...
```
whereas with GCC 7.2.0 (ubuntu-17.10) the output is:
```
>0_1 # 1 # 1344 # 1 # ID=1_1;partial=10;start_type=Edge;rbs_motif=None;rbs_spacer=None;gc_cont=0.612
GTGGATAGCGGAGAGGTGAATCACTTGGAAGACCTGAAGGTCCTGTGGAACGAGATACTGGACAAGGGCA
...
```

The source of the problem is:

https://github.com/hyattpd/Prodigal/blob/004218fbeb2cf54fe12d88ec2eede6a27d70ebf7/gene.c#L312

According to [some sources](https://stackoverflow.com/questions/1973572/can-the-input-and-output-strings-to-sprintf-be-the-same), reading and writing to the same array can lead to undefined behavior.

The following code demonstrates the issue.
```
#### code.c ####

#include <stdio.h>

int main() {

  FILE *ptr = stdout;
  char data[500] = {0};

  sprintf(data, "%sNOTVISIBLE ", data);
  sprintf(data, "%sVISIBLE", data);

  fprintf(ptr, "%s\n", data);

  return 1;

}

#### Makefile ####

SHELL   = /bin/sh
CC      = gcc

CFLAGS  += -pedantic -Wall -O3 -static
LFLAGS = -lm $(LDFLAGS)

TARGET  = test
SOURCES = $(shell echo *.c)
OBJECTS = $(SOURCES:.c=.o)

all: $(TARGET)

$(TARGET): $(OBJECTS)
	$(CC) $(CFLAGS) -o $@ $^ $(LFLAGS)

%.o: %.c
	$(CC) $(CFLAGS) -c -o $@ $<

clean:
	-rm -f $(OBJECTS)
	-rm -f $(TARGET)

.PHONY: all clean
```

When compiled with GCC versions 4.9.3 and 6.4.0 I get `VISIBLE` whereas with GCC 7.2.0 I get `NOTVISIBLE VISIBLE`.

**Comment by tseemann** [comment]
Looks like this issue was fixed and can be closed.

**Comment by unode** [comment]
The pull request is still not merged, so not fixed yet.

**Comment by unode** [comment]
Fixed in #35
